### PR TITLE
Baseline Aligned Row

### LINKED ISSUE #24705: Crash when aligning TextField by baseline
I would like to have a TextField and a button in a Row together, aligned by baseline.  This causes a crash due to an unimplemented function in `package:flutter/src/material/input_decorator.dart`.

```dart
import 'package:flutter/material.dart';

void main() => runApp(Main());

class Main extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Attendance',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(),
        body: Row(
          crossAxisAlignment: CrossAxisAlignment.baseline,
          textBaseline: TextBaseline.alphabetic,
          children: [
            Expanded(child:TextField()),
            Container(width: 16.0),
            RaisedButton(child: Text('REFRESH')),
          ],
        ),
      ),
    );
  }
}
```
Output:

```
I/flutter (25288): ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (25288): The following assertion was thrown during performLayout():
I/flutter (25288): not implemented
I/flutter (25288): 'package:flutter/src/material/input_decorator.dart': Failed assertion: line 966 pos 12: 'false'
```

flutter doctor -v:
```
[√] Flutter (Channel beta, v0.11.9, on Microsoft Windows [Version 10.0.17134.407], locale en-US)
    • Flutter version 0.11.9 at C:\Users\Admin\Desktop\flutter
    • Framework revision d48e6e433c (4 days ago), 2018-11-20 22:05:23 -0500
    • Engine revision 5c8147450d
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at C:\Users\Admin\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[√] Android Studio (version 3.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 30.0.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[√] Connected device (1 available)
    • Pixel XL • HT7470204168 • android-arm64 • Android 9 (API 28)

• No issues found!
```

### PR DESCRIPTION
**Pull request by justinmc**
Closes https://github.com/flutter/flutter/issues/24705

There was a crash when putting a TextField in a row aligned by baseline.  I cherry picked the fix for this part from @GaryQian's upcoming PR https://github.com/flutter/flutter/pull/30069.

Also, the height of the row was not being calculated correctly when aligned by baseline, but is fixed in this PR.

For example, the very tall `counter` on this text field overflows the red Row container.  With this PR, it will fit perfectly inside.
Screen Shot 2019-04-08 at 10 58 49 AM

### CODE REVIEW

File: packages/flutter/test/widgets/basic_test.dart [hunk 1]
```diff
@@ -146,6 +147,44 @@ void main() {
     });
   });
 
+  group('Row', () {
+    testWidgets('multiple baseline aligned children', (WidgetTester tester) async {
+      final GlobalKey key1 = GlobalKey();
```
Comment (HansMuller) [inline packages/flutter/test/widgets/basic_test.dart]: GlobalKeys shouldn't be needed here, `Key key1 = UniqueKey()` should be enough.

File: packages/flutter/test/widgets/basic_test.dart [hunk 2]
```diff
@@ -146,6 +147,44 @@ void main() {
...
+
+      // The two Texts are baseline aligned, so some portion of them extends
+      // both above and below the baseline. The first has a huge font size, so
+      // it extends higher above the baseline than usual. The second has many
+      // lines, but being aligned by the first line's baseline, they hang far
+      // below the baseline. The size of the parent row is just enough to
+      // contain both of them.
+      expect(rowBox.size.height, greaterThan(textBox1.size.height));
+      expect(rowBox.size.height, greaterThan(textBox2.size.height));
+      expect(rowBox.size.height, lessThan(textBox1.size.height + textBox2.size.height));
```
Comment (HansMuller) [inline packages/flutter/test/widgets/basic_test.dart]: We should be able to predict the row's exact overall height and the top or bottom edge of the two text widgets.

https://web-platform-tests.org/writing-tests/ahem.html

**Review by HansMuller — COMMENTED** [review]
This is a really nice fix, and long overdue! The analyzer wants some const expressions and I think the test could be tightened up just a little.

File: packages/flutter/test/widgets/basic_test.dart (hunk 2 above)
Comment (justinmc) [reply packages/flutter/test/widgets/basic_test.dart]: You're right, I was able to calculate the right height.

**Comment by justinmc** [comment]
@HansMuller Ready for re-review.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**